sycl/doc/EnvironmentVariables.md

```diff
@@ -23,6 +23,9 @@ compiler and runtime.
 | `SYCL_ENABLE_DEFAULT_CONTEXTS` | '1' or '0' | Enable ('1') or disable ('0') creation of default platform contexts in SYCL runtime. The default context for each platform contains all devices in the platform. Refer to [Platform Default Contexts](extensions/supported/sycl_ext_oneapi_default_context.asciidoc) extension to learn more. Enabled by default on Linux and disabled on Windows. |
 | `SYCL_RT_WARNING_LEVEL` | Positive integer | The higher warning level is used the more warnings and performance hints the runtime library may print. Default value is '0', which means no warning/hint messages from the runtime library are allowed. The value '1' enables performance warnings from device runtime/codegen. The values greater than 1 are reserved for future use. |
 | `SYCL_USM_HOSTPTR_IMPORT` | Integer | Enable by specifying non-zero value. Buffers created with a host pointer will result in host data promotion to USM, improving data transfer performance. To use this feature, also set SYCL_HOST_UNIFIED_MEMORY=1. |
+| `SYCL_REDUCTION_MAX_WORKGROUP_SIZE_CPU` | Positive integer | Controls the maximum work-group size for reductions on CPU. If `info::device::max_work_group_size` on a device on which a reduction is being enqueued on is less than the value of this environment variable, that value is used instead. If this environment variable is set to 0 or an invalid value, a `sycl::exception` with `sycl::errc::invalid` is thrown during submission of the kernel with a reduction on CPU. If this is unset the maxiumum work-group size for reductions on CPU is implementation defined. |
+| `SYCL_REDUCTION_MAX_WORKGROUP_SIZE_GPU` | Positive integer | Controls the maximum work-group size for reductions on GPU. If `info::device::max_work_group_size` on a device on which a reduction is being enqueued on is less than the value of this environment variable, that value is used instead. If this environment variable is set to 0 or an invalid value, a `sycl::exception` with `sycl::errc::invalid` is thrown during submission of the kernel with a reduction on GPU. If this is unset the maxiumum work-group size for reductions on GPU is implementation defined. |
+| `SYCL_REDUCTION_MAX_WORKGROUP_SIZE_ACC` | Positive integer | Controls the maximum work-group size for reductions on accelerators. If `info::device::max_work_group_size` on a device on which a reduction is being enqueued on is less than the value of this environment variable, that value is used instead. If this environment variable is set to 0 or an invalid value, a `sycl::exception` with `sycl::errc::invalid` is thrown during submission of the kernel with a reduction on accelerator. If this is unset the maxiumum work-group size for reductions on accelerators is implementation defined. |
```

Comment:
misprint: 'maxiumum'.

```suggestion
| `SYCL_REDUCTION_MAX_WORKGROUP_SIZE_CPU` | Positive integer | Controls the maximum work-group size for reductions on CPU. If `info::device::max_work_group_size` on a device on which a reduction is being enqueued on is less than the value of this environment variable, that value is used instead. If this environment variable is set to 0 or a negative  value, a `sycl::exception` with `sycl::errc::invalid` is thrown during submission of the kernel with a reduction on CPU. If this is unset the maximum work-group size for reductions on CPU is implementation defined. |
| `SYCL_REDUCTION_MAX_WORKGROUP_SIZE_GPU` | Positive integer | Controls the maximum work-group size for reductions on GPU. If `info::device::max_work_group_size` on a device on which a reduction is being enqueued on is less than the value of this environment variable, that value is used instead. If this environment variable is set to 0 or a negative value, a `sycl::exception` with `sycl::errc::invalid` is thrown during submission of the kernel with a reduction on GPU. If this is unset the maximum work-group size for reductions on GPU is implementation defined. |
| `SYCL_REDUCTION_MAX_WORKGROUP_SIZE_ACC` | Positive integer | Controls the maximum work-group size for reductions on accelerators. If `info::device::max_work_group_size` on a device on which a reduction is being enqueued on is less than the value of this environment variable, that value is used instead. If this environment variable is set to 0 or a negative value, a `sycl::exception` with `sycl::errc::invalid` is thrown during submission of the kernel with a reduction on accelerator. If this is unset the maximum work-group size for reductions on accelerators is implementation defined. |
```

Reply:
I have fixed the typo, but the intention of "an invalid" is to both include negative numbers and bogus values, for example if someone sets one of these to something that is not a number.

Reply:
Right, Thank you!